packages/plus/src/components/CodeEditor/CodeEditor.tsx

```diff
@@ -2,54 +2,142 @@ import styled from '@emotion/styled'
 import { langs } from '@uiw/codemirror-extensions-langs'
 import { material } from '@uiw/codemirror-theme-material'
 import CodeMirror from '@uiw/react-codemirror'
-import { Text } from '@ultraviolet/ui'
-import type { ComponentProps } from 'react'
+import { consoleDarkTheme } from '@ultraviolet/themes'
+import { CopyButton, Label, Stack, Text } from '@ultraviolet/ui'
+import type { ComponentProps, ReactNode } from 'react'
 
-const StyledText = styled(Text)`
-  background-color: ${({ theme }) => theme.colors.neutral.backgroundStrong};
-  padding: ${({ theme }) => `${theme.space['1']} ${theme.space['2']}`};
-  border-radius: ${({ theme }) => `${theme.radii.default}`};
+const EditorContainer = styled.div`
+  position: relative;
   width: 100%;
 `
 
+const StyledCopyButton = styled(CopyButton)`
+  position: absolute;
+  top: ${({ theme }) => theme.space['1']};
+  right: ${({ theme }) => theme.space['1']};
+  z-index: 1;
+
+  svg > path {
+    fill: ${consoleDarkTheme.colors.other.icon.product.original.fill};
+  }
+
+  &:hover {
+    background-color: ${consoleDarkTheme.colors.neutral.backgroundHover};
+  }
+`
+
+const StyledStack = styled(Stack, {
+  shouldForwardProp: prop => !['disabled'].includes(prop),
+})<{ disabled: boolean }>`
+  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'auto')};
+
+  .cm-editor {
+    font-family: 'JetBrains Mono', monospace;
+    font-size: 14px;
+    background-color: ${({ disabled }) => (disabled ? consoleDarkTheme.colors.neutral.backgroundWeakDisabled : consoleDarkTheme.colors.neutral.backgroundWeak)};
+    border-radius: ${({ theme }) => theme.space['0.5']};
+
+    ${({ disabled }) =>
+      disabled
+        ? `
+      pointer-events: none;
+      color: ${consoleDarkTheme.colors.neutral.textDisabled}`
+        : ''}
+   }
+
+  .cm-content {
+    padding-block: ${({ theme }) => theme.space['2']};
+  };
+
+  .cm-gutters {
+    background-color: ${consoleDarkTheme.colors.neutral.backgroundHover};
+  }
+
+  .cm-line span {
+    ${({ disabled }) =>
+      disabled
+        ? `
+      color: ${consoleDarkTheme.colors.neutral.textDisabled}`
+        : ''}
+  }
+
+  .cm-lineNumbers {
+    padding-left: ${({ theme }) => theme.space['1']};
+  }
+
+  .cm-gutterElement {
+    color: ${({ disabled }) => (disabled ? consoleDarkTheme.colors.neutral.textWeakDisabled : '#545454')};
+  }
+
+  .cm-scroller {
+    border-radius: ${({ theme }) => theme.space['0.5']};
+  }
+`
 type CodeEditorProps = {
-  title?: string
```

Comment:
keep it but mark it as deprecated instead. We don't want to have a breaking change (yet)